src/main/java/sayo/Sayo.java

```diff
@@ -0,0 +1,89 @@
+package sayo;
+
+import java.util.Scanner;
+
+public class Sayo {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        Task[] items = new Task[100];
+        int itemsCount = 0;
+
+        System.out.println("Hello! I'm Sayo and it's great to see you! \n" + "What can I do for you today? \n");
+
+        String input = " ";
+
+        do {
+            try {
+                input = scanner.nextLine().trim();
+                if (input.equals("list")) {
+                    for (int i = 0; i < itemsCount; i++) {
+                        System.out.println((i + 1) + ". " + items[i]);
+                    }
+                } else if (input.startsWith("mark")) {
+                    System.out.println("test");
+                    int index = Integer.parseInt(input.substring(5)) - 1;
+                    if (index >= 0 && index < itemsCount) {
+                        items[index].markAsDone();
+                        System.out.println("Awesome! I've marked this task as done: ");
+                        System.out.println(items[index]);
+                    }
+                } else if (input.startsWith("unmark")) {
+                    int index = Integer.parseInt(input.substring(7)) - 1;
+                    if (index >= 0 && index < itemsCount) {
+                        items[index].unmarkAsDone();
+                        System.out.println("Awesome! I've unmarked this task: ");
+                        System.out.println(items[index]);
+                    }
+                } else if (input.startsWith("todo")) {
+                    itemsCount = addToDo(input, items, itemsCount);
+                } else if (input.startsWith("deadline")) {
+                    int byIndex = input.indexOf("/by");
+                    String description = input.substring(9, byIndex).trim();
+                    String by = input.substring(byIndex + 4).trim();
+                    items[itemsCount] = new Deadline(description, by);
+                    itemsCount++;
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + items[itemsCount - 1]);
+                    System.out.println("Now you have " + itemsCount + " tasks in the list.");
```

Comment:
Perhaps you can abstract out the cases to shorten the method similar to your addToDo method.